operator/controllers/btpoperator_controller_test.go

```diff
@@ -46,102 +50,123 @@ func newFakeK8s(c client.Client) *fakeK8s {
+					Eventually(getCurrentCrState).WithTimeout(stateChangeTimeout).WithPolling(crStatePollingIntevral).Should(Equal(types.StateError))
+				})
 			})
-		})
 
-		Context("When the required Secret is present and all it's data is correct", func() {
-			It("should provision BTP Service Operator successfully", func() {
-				secret, err := createSecretFromYaml()
-				Expect(err).To(BeNil())
-				Expect(k8sClient.Create(ctx, secret)).To(Succeed())
-				Expect(k8sClient.Create(ctx, cr)).To(Succeed())
-				Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: testNamespace, Name: btpOperatorName}, cr)).To(Succeed())
-				Eventually(cr.GetStatus().State, time.Second*30, time.Second*1).Should(Equal(types.StateReady))
+			When("the required Secret's keys do not have all values", func() {
+				It("should return error while verifying values", func() {
+					secret, err := createSecretWithoutValues()
+					Expect(err).To(BeNil())
+					Expect(k8sClient.Create(ctx, secret)).Should(Succeed())
+					Eventually(getCurrentCrState).WithTimeout(stateChangeTimeout).WithPolling(crStatePollingIntevral).Should(Equal(types.StateProcessing))
+					Eventually(getCurrentCrState).WithTimeout(stateChangeTimeout).WithPolling(crStatePollingIntevral).Should(Equal(types.StateError))
+				})
 			})
-		})
 
+			When("the required Secret is correct", func() {
+				It("should install chart successfully", func() {
+					// requires real cluster, envtest doesn't start kube-controller-manager
+					// see: https://book.kubebuilder.io/reference/envtest.html#configuring-envtest-for-integration-tests
+					//      https://book.kubebuilder.io/reference/envtest.html#testing-considerations
+					secret, err := createCorrectSecretFromYaml()
+					Expect(err).To(BeNil())
+					Eventually(k8sClient.Create(ctx, secret)).Should(Succeed())
+					Eventually(getCurrentCrState).WithTimeout(stateChangeTimeout).WithPolling(crStatePollingIntevral).Should(Equal(types.StateReady))
+					btpServiceOperatorDeployment := &appsv1.Deployment{}
+					Eventually(k8sClient.Get(ctx, client.ObjectKey{Name: deploymentName, Namespace: kymaNamespace}, btpServiceOperatorDeployment)).
+						WithTimeout(testTimeout).
+						WithPolling(operationPollingInterval).
+						Should(Succeed())
+				})
+			})
+
+		})
 	})
 
 	Describe("Deprovisioning", func() {
-		BeforeEach(func() {
+		BeforeAll(func() {
 			createSecret()
+		})
 
-			btpOperator := createBtpOperator()
-
-			err := k8sClient.Create(ctx, btpOperator)
-			Expect(err).To(BeNil())
-
-			time.Sleep(time.Second * 30)
+		BeforeEach(func() {
+			cr := createBtpOperator()
+			Expect(k8sClient.Create(ctx, cr)).To(Succeed())
+			Eventually(getCurrentCrState).WithTimeout(stateChangeTimeout).WithPolling(crStatePollingIntevral).Should(Equal(types.StateReady))
 
-			err = clearWebhooks()
+			// required for correct webhooks deletion
+			time.Sleep(time.Second * 1)
```

Comment:
Remove that sleep and clearing webhooks. We waste 1sec per test. To solve that issue add two additional mandatory spec params to SI and SB and webhooks will not complain.

Reply:
we discussed the matter and didn't find any good solution for webhooks behaviour atm